src/core/sprites/webgl/SpriteRenderer.js

```diff
@@ -108,16 +109,13 @@ export default class SpriteRenderer extends ObjectRenderer
         // step 2: check the maximum number of if statements the shader can have too..
         this.MAX_TEXTURES = checkMaxIfStatmentsInShader(this.MAX_TEXTURES, gl);
 
-        this.shaders = new Array(this.MAX_TEXTURES);
-        this.shaders[0] = generateMultiTextureShader(gl, 1);
-        this.shaders[1] = generateMultiTextureShader(gl, 2);
+        const shader = this.shader = generateMultiTextureShader(gl, this.MAX_TEXTURES);
```

Comment:
As long as the branching is consistent per pixel it should be fine. Its when you actually do get up to using multiple textures (4, 5, 6, ...) that the branching becomes a concern because it will be very different and the parallelism of the GPU will be shot.

> where it complained about using textures that are not bound

Is that why you stuff dummy textures into the state, so that when this shader runs there are always texture sin each slot even if we don't use them?